fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -616,25 +623,45 @@ private List<String> checkAndExtractPartitionCol(FunctionCallExpr expr) {
                 throw new ParsingException(PARSER_ERROR_MSG.unsupportedExprWithInfo(expr.toSql(), "PARTITION BY"), pos);
             }
             Expr firstExpr = paramsExpr.get(0);
+            Expr secondExpr = paramsExpr.get(1);
+            String partitionName;
+            if (secondExpr instanceof SlotRef) {
+                partitionName = ((SlotRef) secondExpr).getColumnName();
```

Comment:
```suggestion
                partitionColumnName = ((SlotRef) secondExpr).getColumnName();
```

Reply:
better to put the check logic of DATA_TRUNC and TIME_SLICE into a function each.